src/query/RelayQuery.js

```diff
@@ -214,6 +218,36 @@ class RelayQueryNode {
     return children;
   }
 
+  shouldBeIncluded(): ?boolean {
+    var directives = this.getDirectives();
+
+    var skipNode = directives.find(
+      directive => directive.name === SKIP
+    );
+    if (skipNode) {
+      var skipIf = skipNode.arguments.find(
+        arg => arg.name === IF
+      );
+      if (skipIf) {
+        return !skipIf.value;
+      }
+    }
+
+    var includeNode = directives.find(
+      directive => directive.name === INCLUDE
+    );
+    if (includeNode) {
+      var includeIf = includeNode.arguments.find(
+        arg => arg.name === IF
+      );
+      if (includeIf) {
+        return Boolean(includeIf.value);
+      }
+    }
+
+    return true
```

Comment:
Because the schema may change, it's definitely good to scan the arguments for each directive. To make this more performant, how about iterating over the directives only once? Something like

```javascript
let isSkipped = false;
let isIncluded = true;
directives.forEach(directive => {
  if (directive.name === SKIP) {
    isSkipped = ...
  } else if (directive.name === INCLUDE) {
    isIncluded = ...
  }
});
return isSkipped || !isIncluded;
```